networking/Field_Control_Networking_Spec.md

```diff
@@ -0,0 +1,83 @@
+# Field Control Networking Specificiation
+
+## Contents
+
+1. Overview
+2. Setting up the Clients and Network
+3. Connecting to the Field Control Network
+4. Establishing Communication Links
+5. Sending Messages
+
+## 1. Overview
+
+A robot interacts with two networks:  the Student network and the Field Control (FC) network.  The former is the network over which students communicate with their robot via provided mini-routers.  The latter--the focus of this spec--is the network over which students' robots interact with FC over the course of a match.
+
+Clients (field control, driver stations, and robots) are assigned static IPs and communicate over TCP connections.
+
+FC sends out messages to initiate robot-to-driver-station pairing and robot mode changes.  FC listens to heartbeat messages from robots.
+
+## 2. Setting Up the Clients and Network
+
+### Clients
+
+The FC network will have up to eight clients, as follows:
+
+- One FC client.  This is the FC computer running the match.  A PiE computer+monitor. 
+- Four driver station clients.  This acts as the student-robot interface (via Dawn and Runtime), and as the robot-field control interface (see "Establishing Communication Links").  A wifi-enabled laptop or  computer+monitor.
+- Up to four robot clients.  A wifi-enabled Beagle Bone Black (via wifi dongle).  One per student robot.
```

Comment:
In addition to video streaming, the match schedule used to be pulled from an online source (e.g. a Google Spreadsheet), such that both event planners and field control operate off the same source of ground truth. I think one year we passed around schedules on paper slips and USB drives... that was a mess.

This also adds the need for a computer that has access to both field control and the internet.